### PR TITLE
Add conda build check on CircleCI

### LINKED ISSUE #3133: Employ CI build and test systems for Python API
A CI build/test system would
1) Document the steps for building the Python API
2) Add checks to see if PR's break the API
3) May allow for more comprehensive unit testing (no graphics cards on Circle nodes though...)

### PR DESCRIPTION
**Pull request by sgpearse**
Fixes #3133, by adding a compilation check for the conda .tar.bz2 package.  This tar.bz2 file, and all intermediate compilation files are stored as artifacts on CircleCI.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -335,10 +334,63 @@ jobs:
...
-  build_ubuntu18:
+  python_api_centos:
+    docker:
+      - image: sgpearse/vapor3-centos7:latest
+
+    resource_class: large
+
+    steps:
+      - run:
+          name: update 3rd party libraries
```
Comment (StasJ) [inline .circleci/config.yml:345]: The conda build does not use the third party libraries.

File: .circleci/config.yml [hunk 2]
```diff
@@ -335,10 +334,63 @@ jobs:
...
+          command: |
+            rm -rf /usr/local/VAPOR-Deps/2019-Aug
+            pip3 install gdown --upgrade
+            gdown https://drive.google.com/uc?id=1S9DwySMnQrBuUUZGKolD__WQrjTmLgyn
+            filename="/root/project/2019-Aug-CentOS.tar.xz"
+            bsdtar -xf ${filename} -C /usr/local/VAPOR-Deps
+
+      - run:
+          name: install python
```
Comment (StasJ) [inline .circleci/config.yml:354]: Python is provided by conda.

File: .circleci/config.yml [hunk 3]
```diff
@@ -335,10 +334,63 @@ jobs:
...
+          command: |
+            yum -y update || true
+            yum install -y python3
+            yum install -y python3-pip
+            pip3 install --upgrade pip
+            pip3 install scikit-build command
+
+      - run:
+          name: install miniconda
```
Comment (StasJ) [inline .circleci/config.yml:363]: You could just use the miniconda docker image.

File: .circleci/config.yml [hunk 4]
```diff
@@ -335,10 +334,63 @@ jobs:
...
+            yum install -y wget
+            wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh
+            bash ~/miniconda.sh -b -p ~/miniconda 
+
+      - checkout
+
+      - run:
+          name: conda build .
+          command: |
+            yum -y install doxygen
```
Comment (StasJ) [inline .circleci/config.yml:374]: DOxygen is provided by conda.

File: .circleci/config.yml [hunk 5]
```diff
@@ -335,10 +334,63 @@ jobs:
...
+            rm -rf /root/project/build/*
+            rm -f /root/project/site.local
+            cd /root/project/build
+            cd /root/project/conda
+            export PYTHONPATH=/root/miniconda/lib/python3.9
```
Comment (StasJ) [inline .circleci/config.yml:379]: This is non-portable with regards to conda and is handled by the build anyway.

File: .circleci/config.yml [hunk 6]
```diff
@@ -335,10 +334,63 @@ jobs:
...
+            /root/miniconda/bin/conda install -y conda-build
+            /root/miniconda/bin/conda config --add channels conda-forge
+            /root/miniconda/bin/conda build .
+          no_output_timeout: 20m
+ 
+      - store_artifacts:
+          path: /root/miniconda/conda-bld/linux-64
```
Comment (StasJ) [inline .circleci/config.yml:386]: We only need the tarball.

File: .circleci/config.yml [hunk 7]
```diff
@@ -335,10 +334,63 @@ jobs:
...
+
+  python_api_ubuntu:
```
Comment (StasJ) [inline .circleci/config.yml:388]: The conda build generates a universal linux binary.

File: .circleci/config.yml [hunk 8]
```diff
@@ -350,15 +402,55 @@ jobs:
             pip3 install scikit-build command
 
       - run:
-          name: update 3rd party libraries
+          name: update cmake
```
Comment (StasJ) [inline .circleci/config.yml:405]: cmake is provided by conda.

**Review by StasJ — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 3 above)
Comment (sgpearse) [reply .circleci/config.yml:345]: I didn't know they had one based on debian.  Sounds good.

File: .circleci/config.yml (hunk 4 above)
Comment (sgpearse) [reply .circleci/config.yml]: Good to know.

File: .circleci/config.yml (hunk 6 above)
Comment (sgpearse) [reply .circleci/config.yml]: CircleCI is finicky about how it uploads artifacts, in that wildcards can't be used to select files.  Only directories can be targeted.  I'll just move the tarball to an empty directory with the wildcard, and then target that directory.

File: .circleci/config.yml (hunk 7 above)
Comment (sgpearse) [reply .circleci/config.yml:369]: Would you agree that it's still beneficial to have documented build rules for each platform?

File: .circleci/config.yml (hunk 8 above)
Comment (sgpearse) [reply .circleci/config.yml]: Got it.

File: .circleci/config.yml (hunk 5 above)
Comment (sgpearse) [reply .circleci/config.yml]: Fixed.

**Comment by sgpearse** [comment]
@stasj - Thanks for the recommendations.  They simplify things quite a bit.  

Everything should be addressed except for the comment regarding the linux builds generating the same installer.  I'm thinking that it would be good to have build rules for all *nix platforms so we can trigger one-off builds if we need an installer.  I've added a new job for OSX (python_api_osx) for this.  The centOS and OSX builds are currently turned off, but you can see their runs succeed in commit [c112cac](https://github.com/NCAR/VAPOR/pull/3187/commits/c112cac926c6f0d64d29fb37c3311d983076530b).

File: .circleci/config.yml (hunk 1 above)
Comment (sgpearse) [reply .circleci/config.yml]: Huh I thought I commented here.  Fixed.

**Pull request merged**